cocos/2d/framework/ui-component.ts

```diff
@@ -66,8 +67,9 @@ export class UIComponent extends Component {
         // TODO: UIComponent should not be assigned to UIMeshRenderer | UIRenderer @holycanvas
         // workaround: mark this as any
         // issue: https://github.com/cocos/cocos-engine/issues/14637
-        if ((this as any).node._uiProps.uiComp === this) {
-            (this as any).node._uiProps.uiComp = null;
+        const uiProps = this.node._uiProps;
+        if ((uiProps.uiComp as unknown as UIComponent) === this) {
```

Comment:
Why convert to UIComponent here and convert to UIRenderer above?

Reply:
Fixed.